src/test/java/com/books/services/BookServiceTest.java

```diff
@@ -32,13 +32,13 @@ public void setUp() {
                 new Person(4,"Дмитрий", "Fourth")
         };
         books = new Book[]{
-                new Book(0,"first book", date, new Publisher("Publisher1"),
+                new Book(0,"first book", date, new Publisher(0,"Publisher1"),
```

Comment:
Define Constrctor without id param in Publisher class. And as I said previously for now identity management is responsibility of database